cats/secure_the_bag.py

```diff
@@ -0,0 +1,106 @@
+from typing import Dict, List, Tuple, Union
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.util.hash import std_hash
+from chia.util.ints import uint64
+from chia.wallet.cat_wallet.cat_utils import construct_cat_puzzle
+from chia.wallet.puzzles.cat_loader import CAT_MOD
+from clvm.casts import int_to_bytes
+
+# Fees spend asserts this. Message not required as inner puzzle contains hardcoded coin spends and doesn't accept a solution.
+EMPTY_COIN_ANNOUNCEMENT = [ConditionOpcode.CREATE_COIN_ANNOUNCEMENT, None]
+
+
+class Target:
+    puzzle_hash: bytes32
+    amount: uint64
+
+    def __init__(self, puzzle_hash: bytes32, amount: uint64) -> None:
+        self.puzzle_hash = puzzle_hash
+        self.amount = amount
+    
+    def create_coin_condition(self) -> Tuple[bytes, bytes32, uint64, Tuple[bytes32]]:
+        return [ConditionOpcode.CREATE_COIN, self.puzzle_hash, self.amount, [self.puzzle_hash]]
```

Comment:
it looks like the types got mixed up here, this returns a list but should return a Tuple.

what is the reason for that last Tuple[Byte32]  ?

Reply:
The `Tuple[bytes32]` is the hint containing the inner puzzle hash for full nodes in index.